tests/ui/test_ui.py

```diff
@@ -81,6 +81,29 @@ def test_set_footer_text_with_duration(self, view, mocker,
         mock_sleep.assert_called_once_with(duration)
         assert view.controller.update_screen.call_count == 2
 
+    @pytest.mark.parametrize('state, footer_text', [
+        (0, [('code', ' help '), ' community ', ' issues ']),
+        (1, [' help ', ('code', ' community '), ' issues ']),
+        (2, [' help ', ' community ', ('code', ' issues ')]),
+        (3, [' help ', ' community ', ' issues ']),
+        (None, [' help ', ' community ', ' issues ']),
+        (30, [' help ', ' community ', ' issues ']),
+    ])
+    def test_set_typeahead_footer(self, mocker, view, suggestions,
+                                  state, footer_text):
+        set_footer_text = mocker.patch('zulipterminal.ui.View.set_footer_text')
+        view.set_typeahead_footer(suggestions[:3], state)
```

Comment:
You can parameterize the number of suggestions rather than hard-coding (`3`) it here to make it more readable. Also, you might want to add a little variation in the number of suggestions along with that.